kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -141,8 +146,13 @@ def __init__(  # noqa: PLR0913
 
         self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
         if save_args is not None:
+            if table_args is not None:
+                save_args["database"] = table_args.get("database", None)
```

Comment:
This feels a bit magical to me. It's not really consistent with the docstring, either, which says that arguments will be passed to `create_{materialized}`; in reality, the user needs to know that just `database` will be passed.

I personally would recommend one of two approaches. One is to not do anything special here; the user can pass `database` in `save_args` and `database` in `table_args`, and, while it may feel duplicative, at least it's explicit. The other approach to make an explicit `database` keyword for the dataset, and likely raise an error if `database` is specified in `save_args` and/or `table_args` if also passed explicitly.

@mark-druffel does this make sense, and do you have a preference?

Reply:
@deepyaman As discussed yesterday, I've moved database to the top-level as discussed.  I'm trying to push the changes, but I'm getting blocked by pre-commit now that I have it setup properly. 

When it ran, it changed a bunch of files I never touched. I staged those as well (not sure if I should've), but my commit still failed because of Black. I've run black manually on the file I changed too to try to lint the file. Any suggestions how I can get this working properly? :grimacing: 

![image](https://github.com/user-attachments/assets/94b397cc-7263-4eaf-871f-0405a5cc59ee)